### ISSUE TITLE
Incorrect reduction axis in C51 loss when `critic_use_cdq=False`

### ISSUE THREAD

**Issue by wzhhasadream** [DAVIAN-Robotics/V-Simba#2]
Hi, I noticed a potential reduction-axis issue in `categorical_td_loss` in `scale_rl/agents/vsimba/vsimba_update.py`.

When `critic_use_cdq=False`, `VSimbaCritic` still uses `nn.vmap` with `out_axes=0`, so `pred_log_probs` retains the singleton critic dimension:

```python
pred_log_probs.shape == (1, batch_size, num_bins)
target_probs.shape == (batch_size, num_bins)
```

However, the loss is computed as:

```python
loss = -jnp.mean(jnp.sum(target_probs * pred_log_probs, axis=1))
```

After broadcasting, `axis=1` is the **batch dimension**, rather than the categorical support dimension. Consequently, the loss and its gradients are scaled by `batch_size / num_bins` relative to the standard batch-averaged cross-entropy. With the default values, this factor is `256 / 101 ≈ 2.535`. I confirmed this scaling numerically.

The double-critic branch is unaffected because it indexes each head before calling this function, producing `(batch_size, num_bins)` tensors.

A possible fix is to sum over the last dimension:

```diff
- loss = -jnp.mean(jnp.sum(target_probs * pred_log_probs, axis=1))
+ loss = -jnp.mean(jnp.sum(target_probs * pred_log_probs, axis=-1))
```

This handles both shapes consistently. The issue changes the loss scale, not the sample–target pairing. I have not evaluated its effect on training performance, and AdamW may largely compensate for constant gradient scaling.

Could you confirm whether this scaling was intentional?
